core/src/main/java/org/apache/brooklyn/core/mgmt/ha/OsgiManager.java

```diff
@@ -189,10 +193,11 @@ public static boolean isBundleNameEqualOrAbsent(CatalogBundle bundle, Bundle b)
                 Maybe<Bundle> bundle = findBundle(osgiBundle);
                 if (bundle.isPresent()) {
                     Bundle b = bundle.get();
-                    Class<T> clazz;
+                    Optional<String> strippedType = osgiClassPrefixer.stripMatchingPrefix(b, type);
+                    String typeToLoad = strippedType.isPresent() ? strippedType.get() : type;
```

Comment:
Not sure if this is the right place for the fix! Arguably if it has the prefix, we should only look in one bundle (or multiple versions of the bundle), rather than iterating over all bundles.